Bookish/Bookish.Web/Models/CatalogueViewModel.cs

```diff
@@ -11,22 +11,51 @@ public class CatalogueViewModel
         public IEnumerable<Book> Books { get; }
         public string SearchTerm { get; }
         public int PageNumber { get; }
+        public int PageItems { get; }
+        public int MaxPages { get; }
 
         public CatalogueViewModel(IEnumerable<Book> books, string searchTerm, int pageNumber)
         {
             Books = books;
             SearchTerm = searchTerm;
-            PageNumber = pageNumber;
+            PageItems = 5;
+            MaxPages = (Books.Count() + PageItems - 1) / PageItems;
+            PageNumber = FixPageNumber(pageNumber);
         }
 
-        public string GetDisabledClass(bool isNext)
+        public int FixPageNumber(int pageNumber)
         {
-            if (isNext)
+            if (pageNumber < 1)
             {
-                return Books.Count() < 5 ? "disabled" : "";
+                return 1;
             }
 
+            if (pageNumber > MaxPages)
+            {
+                return MaxPages;
+            }
+
+            return pageNumber;
+        }
+
+        public IEnumerable<Book> GetPage()
```

Comment:
Instead of having this as a function you could assign `Books` to the correct books to display on this page? 🙂